subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/BuildMetricsPlugin.kt

```diff
@@ -34,6 +37,9 @@ open class BuildMetricsPlugin : Plugin<Project> {
             SentryConsumer(project),
             buildTraceConsumer(project)
         ))
+        project.tasks.register<CollectTeamcityMetricsTask>("collectTeamcityMetrics") {
+            buildId.set(project.getOptionalStringProperty("avito.build.metrics.teamcityBuildId"))
```

Comment:
I had to move a validation to a task.
Otherwise, it fails `:tasks` in local builds. I found this unexpected and needless.